components/bms_worker/src/CAN/CANIO-tx.c

```diff
@@ -202,66 +203,127 @@ static const packTable_S* packNextMessage(const packTable_S* packTable,
+static bool MSG_pack_BMS_1Hz_Cell_Voltage_6_to_11(CAN_data_T* message, const uint8_t counter)
 {
-    UNUSED(message);
     UNUSED(counter);
-    return false;
+    message->u64  = ((uint64_t)((uint16_t)(BMS.cells[6].voltage / 5) & 0x3FF)) << 54;   //10 bits, 5mV precision, range [0-5120]mV
+    message->u64 |= ((uint64_t)((uint16_t)(BMS.cells[7].voltage / 5) & 0x3FF)) << 44;   //10 bits, 5mV precision, range [0-5120]mV
+    message->u64 |= ((uint64_t)((uint16_t)(BMS.cells[8].voltage / 5) & 0x3FF)) << 34;   //10 bits, 5mV precision, range [0-5120]mV
+    message->u64 |= ((uint64_t)((uint16_t)(BMS.cells[9].voltage / 5) & 0x3FF)) << 24;   //10 bits, 5mV precision, range [0-5120]mV
+    message->u64 |= ((uint64_t)((uint16_t)(BMS.cells[10].voltage / 5) & 0x3FF)) << 14;  //10 bits, 5mV precision, range [0-5120]mV
+    message->u64 |= ((uint64_t)((uint16_t)(BMS.cells[11].voltage / 5) & 0x3FF)) << 4;   //10 bits, 5mV precision, range [0-5120]mV
+    return true; //60 bits used, 4 unused
 }
 
-static bool MSG_pack_BMS_1Hz(CAN_data_T* message, const uint8_t counter)
+static bool MSG_pack_BMS_1Hz_Cell_Voltage_12_to_15(CAN_data_T* message, const uint8_t counter)
 {
-    UNUSED(message);
     UNUSED(counter);
-    return false;
+    message->u64  = ((uint64_t)((uint16_t)(BMS.cells[12].voltage / 5) & 0x3FF)) << 54;  //10 bits, 5mV precision, range [0-5120]mV
+    message->u64 |= ((uint64_t)((uint16_t)(BMS.cells[13].voltage / 5) & 0x3FF)) << 44;  //10 bits, 5mV precision, range [0-5120]mV
+    message->u64 |= ((uint64_t)((uint16_t)(BMS.cells[14].voltage / 5) & 0x3FF)) << 34;  //10 bits, 5mV precision, range [0-5120]mV
+    message->u64 |= ((uint64_t)((uint16_t)(BMS.cells[15].voltage / 5) & 0x3FF)) << 24;  //10 bits, 5mV precision, range [0-5120]mV
+    message->u64 |= ((uint64_t)((uint16_t)(BMS.pack_voltage / 10) & 0x1FFF)) << 11;     //13 bits, 10mV precision, range [0-81910]mV
+    return true; //53 bits used, 11 unused
+}
+
+static bool MSG_pack_BMS_1Hz_Temperatures_and_Humidity(CAN_data_T* message, const uint8_t counter)
+{
+    UNUSED(counter);
+    message->u8[0] = (uint8_t)(ENV.values.board.brd_temp[0] / 10);  //8 bits, 1 deg C precision, range [0-255]deg C
+    message->u8[1] = (uint8_t)(ENV.values.board.brd_temp[1] / 10);  //8 bits, 1 deg C precision, range [0-255]deg C
+    message->u8[2] = (uint8_t)(ENV.values.board.mcu_temp / 10);     //8 bits, 1 deg C precision, range [0-255]deg C
+    message->u8[3] = (uint8_t)(ENV.values.board.ambient_temp / 10); //8 bits, 1 deg C precision, range [0-255]deg C
+    message->u8[4] = (uint8_t)(ENV.values.board.rh / 100);          //8 bits, 1% precision, range [0-255]%
+    return true; //40 bits used, 24 unused
+}
+
+static bool MSG_pack_BMS_1Hz_Fans(CAN_data_T* message, const uint8_t counter)
+{
+    UNUSED(counter);
+    message->u8[0]  = COOL.percentage[0];   //8 bits, 1% precision, range [0-255]%
+    message->u8[1]  = COOL.percentage[1];   //8 bits, 1% precision, range [0-255]%
```

Comment:
percentage has range 0-100, can both be 7 bit but 8 bit is simple wont change anything here. arguably can also have 8th bit as fan on or fan off flag for each fan, so like byte 1 has bits 0-6 as percent and bit 7 as on or off